### ISSUE TITLE
[N64] Ocarina of Time Glitched Subscreen / Inventory

### ISSUE THREAD

**Issue by Sirius902** [TASEmulators/BizHawk#1372]
What we know:
- This bug affects unmodded roms and not romhacks simply because they have different SHA1 checksums. There's nothing special about the roms' contents.
- Normally, game-specific emulation hacks are enabled by flags in the gamedb, but there are some baked into the copy of Mupen64Plus' core distributed with BizHawk.
- We can't recompile the core right now.
- @steventylerseanherr has (unwittingly?) provided a [workaround](https://github.com/TASVideos/BizHawk/issues/1372#issuecomment-485263461): change the rom slightly with a hex editor.

Original bug report follows.

---

The inventory screen is glitched for OOT such that when using GlideN64, the inventory screen is always white and when using Glide64mk2 it looks very broken. I suspect it has something to do with some sort of hard-coded Subscreen Delay Fix conflicting with the graphics plugins' framebuffer emulation since the inventory works fine in OOT Randomizer which isn't recognized as OOT and thus the delay fix isn't applied to it.

GlideN64 OOT
![oot gliden64](https://user-images.githubusercontent.com/10891979/48803824-df12d300-ecc7-11e8-9c96-d86637612e3a.png)

Glide64mk2 OOT
![oot glidemk2](https://user-images.githubusercontent.com/10891979/48803834-e5a14a80-ecc7-11e8-95a9-cf63ee9d6c68.png)

GlideN64 OOT Randomizer
![oot rando gliden64](https://user-images.githubusercontent.com/10891979/48803848-ed60ef00-ecc7-11e8-932e-c19aec0f7efc.png)

**Comment by vadosnaprimer** [comment]
Test in dev build please.

**Comment by Sirius902** [comment]
I tested this in the most recent dev build, currently from 2/22, and it still has the same issue.

**Comment by steventylerseanherr** [comment]
if its still up for testing i can say that the latest release and good dev build does the same
<img width="160" alt="Legend of Zelda, The - Ocarina of Time (USA) 2019-03-19 00 50 56" src="https://user-images.githubusercontent.com/48705919/54581038-33f9f800-49e1-11e9-8eee-ccd590a2911f.png">
seems to be randomish     i can always trigger it on the platform of links house
all 4 graphics plugins each with a simuler effect
and all 3 display methods

i played around with the dev builds enable fb emulation
and with it ticked off i could cause the glitch anywhere

haha just ask for anymore information if needed

**Comment by steventylerseanherr** [comment]
update:
sure enough a randomized rom doesnt have this issue
tested on dev build 2019 03 15 151853

**Comment by steventylerseanherr** [comment]
little more then a month and no response?
guess ill just make another thread then

**Comment by YoshiRulz** [comment]
We're not really able to fix any bugs in Mupen64Plus until we can build it again.

**Comment by steventylerseanherr** [comment]
just tested in more depth
if bizhawk detects a rom with  
........THE LEGEND OF ZELDA .......CZLE
specifically the "THE LEGEND OF ZELDA"
then i get a white background on pausing in a 3d area

if i change that to something else then it works as its supposed to

**Comment by YoshiRulz** [comment]
Wait, it depends on the filename?

**Comment by steventylerseanherr** [comment]
no 
the games title in hex
something to do with rom identification
<img width="715" alt="Capture" src="https://user-images.githubusercontent.com/48705919/56472600-5f339500-642e-11e9-8870-c1bbfabdabb8.PNG">

**Comment by YoshiRulz** [comment]
Right, so BizHawk uses the game's SHA1 checksum to tell games apart. If you change even 1 bit with a hex editor, it's going to think it's a different rom and you won't see a checkmark in the status bar.

**Comment by vadosnaprimer** [comment]
Try tweaking the video plugin settings.

**Comment by steventylerseanherr** [comment]
and when bizhawk detects oot as a different rom then the white background is gone and works as intended
clean oot rom
<img width="326" alt="clean oot rom" src="https://user-images.githubusercontent.com/48705919/56472740-7ecbbd00-6430-11e9-908c-c3e5470fcbfe.PNG">
clean oot rom disguised as something else
<img width="328" alt="clean oot rom disguised as something else" src="https://user-images.githubusercontent.com/48705919/56472744-8db26f80-6430-11e9-9dc8-6bc2f40324d2.PNG">

i will say that with the white background gone that the pause screen delay is back in full force
surely there is a way to fix the white screen glitch and the delay glitch at the same time?

and i dont want to be that guy that does this but
pj64 with gliden64 doesnt do this

**Comment by YoshiRulz** [comment]
Might find this extremely relevant -_-
https://github.com/TASVideos/BizHawk/blob/315e5d64b1a21cd37af14e17bfa2036f11ff59b0/libmupen64plus/mupen64plus-core/src/main/cheat.c#L233-L260

**Comment by steventylerseanherr** [comment]
yep
id say thats the fix for the pause screen lag
using that cheat causes the white screen problem though

**Comment by steventylerseanherr** [comment]
latest nightly pj64 with gliden64 on my pc doesnt have the white screen or the delay at all so im wondering if its possible to fix at all

**Comment by YoshiRulz** [comment]
Because there's nothing wrong with our copy of GLideN64, we'll eventually get it fixed, the problem here is going to be building the Mupen core.

**Comment by steventylerseanherr** [comment]
ok
its always good to know if a problem will be fixed eventually instead of just being ignored

**Comment by Sirius902** [comment]
The issue does not exist when using Mupen64plus from https://m64p.github.io/ but it's likely that they used a newer version of the core.

**Comment by vadosnaprimer** [comment]
Our m64p is from 2013 and no one is able to update it.

**Comment by steventylerseanherr** [comment]
yeah i was wondering about that
why not?

**Comment by vadosnaprimer** [comment]
It requires reapplying all the functionality that hawk needs from the core, and their codebase is totally different now. @NarryG tried without success.

**Comment by Nicholas-Steel** [comment]
I was going to report this too, lucky I searched first. My own observation is it's some GlideN64 configuration being loaded from a internal database for recognized ROM's and there's no way in the UI to override the behaviour.

**Comment by YoshiRulz** [comment]
@Nicholas-Steel As in [my earlier comment](https://github.com/TASVideos/BizHawk/issues/1372#issuecomment-485265651), the specific (broken) hack that only applies to this rom isn't set in the gamedb with the others, but is hardcoded in the Mupen core library that comes with BizHawk. I've edited a summary into the issue desc to make this clear to future humans.

**Comment by Nicholas-Steel** [comment]
No problems, thanks for the summary.
